wgengine/userspace.go

```diff
@@ -208,13 +209,14 @@ func newUserspaceEngineAdvanced(conf EngineConfig) (_ Engine, reterr error) {
 		Forward: true,
 	}
 	e := &userspaceEngine{
-		timeNow:  time.Now,
-		logf:     logf,
-		reqCh:    make(chan struct{}, 1),
-		waitCh:   make(chan struct{}),
-		tundev:   tstun.WrapTUN(logf, conf.TUN),
-		resolver: tsdns.NewResolver(rconf),
-		pingers:  make(map[wgkey.Key]*pinger),
+		timeNow:             time.Now,
+		logf:                logf,
+		reqCh:               make(chan struct{}, 1),
+		waitCh:              make(chan struct{}),
+		tundev:              tstun.WrapTUN(logf, conf.TUN),
+		resolver:            tsdns.NewResolver(rconf),
+		pingers:             make(map[wgkey.Key]*pinger),
+		networkMapCallbacks: make(map[*networkMapCallbackHandle]NetworkMapCallback),
```

Comment:
you could also not initialize it here and do it lazily on the first added callback